### PR TITLE
docs: Add first draft of testing strategy for SplashKit distribution channels

### PR DESCRIPTION
**Pull request by isdampe**
# Description

This PR includes first draft of testing strategy document based on the Thot Tech documentation template.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [x] Documentation (update or new)

# How Has This Been Tested?

Proof read and formatted with prettier.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have requested a review from @talzid and @lmorg42 on the Pull Request

### CODE REVIEW

**Comment by talzid** [comment]
@isdampe were Lachlan and I supposed to be Assigned or Reviewers?

File: docs/Splashkit/Operations/Distribution Channels/testing/test-strategy.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+1. This is a manual process and the responsibility is left with the developer or reviewer to fully
+   test their code. Reviewers may have to set up a Linux environment to test changes.
+
+### **Windows**
+
+[ TBA ]
+
+## **References**
+
+1. [Legality of running macOS in virtualised environments](https://discussions.apple.com/thread/6135949#:~:text=It%20is%20only%20legal%20to,to%20VMware%20Fusion%20and%20Parallels.)
```
Comment (talzid) [inline docs/Splashkit/Operations/Distribution Channels/testing/test-strategy.md:80]: This is the url for the page without the text search extras. Just a bit cleaner

```suggestion
1. [Legality of running macOS in virtualised environments](https://discussions.apple.com/thread/6135949)
```

File: docs/Splashkit/Operations/Distribution Channels/testing/test-strategy.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+SplashKit's management tool `skm`. Given the nature of this work (writing installation scripts),
+there is no fixed testing strategy that available across all three supported operating systems
+(Windows, macOS and Linux). Different tools are available for different operating systems, which is
+dictated by both technology availability as well as licensing issues.
+
+### **macOS**
+
+[macOS cannot be legally run in a virtualised environment](https://discussions.apple.com/thread/6135949#:~:text=It%20is%20only%20legal%20to,to%20VMware%20Fusion%20and%20Parallels.)
+unless the host system is also running on Apple hardware, making installation tests through
+virtualisation impracticle.
```
Comment (kats8) [inline docs/Splashkit/Operations/Distribution Channels/testing/test-strategy.md:15]:
```suggestion
virtualisation impractical.
```

File: docs/Splashkit/Operations/Distribution Channels/testing/test-strategy.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+Whilst [unofficial docker images](https://github.com/sickcodes/Docker-OSX) for macOS exist, they are
+not formally supported by Apple, making containersed testing also unviable.
```
Comment (kats8) [inline docs/Splashkit/Operations/Distribution Channels/testing/test-strategy.md:18]:
```suggestion
not formally supported by Apple, making containerised testing also unviable.
```

**Comment by kats8** [comment]
Thank you for the consideration of testing strategy and challenges for SDK :). I've just marked a couple of typos I noticed while reading the draft.

**Review by lmorg42 — APPROVED** [review]

**Comment by talzid** [comment]
Happy to merge now typos are fixed

**Pull request merged**